libs/azure-storage/README.md

```diff
@@ -1,9 +1,184 @@
+
+loader = AzureBlobStorageLoader(
+    account_url="http://<my-storage-account-name>.blob.core.windows.net",
+    container_name="<my-container-name>",
+    blob_names="<my-pdf-file.pdf>",
+    loader_factory=PyPDFLoader,
+)
+```
+
+To provide additional configuration, you can define a callable that returns an instantiated document loader as shown below:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_community.document_loaders import PyPDFLoader
+
+def loader_factory(file_path: str) -> PyPDFLoader:
+    return PyPDFLoader(
+        file_path,
+        mode="single",  # Custom configuration
+        pages_delimiter="\n",  # Custom configuration
+    )
+
+loader = AzureBlobStorageLoader(
+    account_url="http://<my-storage-account-name>.blob.core.windows.net",
+    container_name="<my-container-name>",
+    blob_names="<my-pdf-file.pdf>",
+    loader_factory=loader_factory,
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+## Migrating from LangChain Community Azure Storage Document Loaders
+This section goes over the actions required to migrate from the existing community document loaders to the new Azure Blob Storage document loader:
+
+1. Depend on the `langchain-azure-storage` package instead of `langchain-community`.
+2. Update import statements from `langchain_community.document_loaders` to
+   `langchain_azure_storage.document_loaders`.
+3. Change class names from `AzureBlobStorageFileLoader` and `AzureBlobStorageContainerLoader`
+   to `AzureBlobStorageLoader`.
+4. Update document loader constructor calls to:
+    1. Use an account URL instead of a connection string.
+    2. Specify `UnstructuredLoader` as the `loader_factory` if they want to continue to use Unstructured for parsing documents.
+5. Ensure environment has proper credentials (e.g., running `azure login` command, setting up managed identity, etc.) as the connection string would have previously contained the credentials.
+
+The examples below show the before and after migrating to the `langchain-azure-storage package`:
+
+#### Before migration
+```python
+from langchain_community.document_loaders import AzureBlobStorageFileLoader, AzureBlobStorageContainerLoader
+
+file_loader = AzureBlobStorageFileLoader(
+    conn_str="<my-connection-string>",
+    container="<my-container-name>",
+    blob_name="<my-blob-name>",
+)
+
+container_loader = AzureBlobStorageContainerLoader(
+    conn_str="<my-connection-string>",
+    container="<my-container-name>",
+    prefix="<prefix>",
+)
+```
+
+#### After migration
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_unstructured import UnstructuredLoader
+
+loader = AzureBlobStorageLoader(
+    account_url="http://<my-storage-account-name>.blob.core.windows.net",
+    container_name="<my-container-name>",
+    loader_factory=UnstructuredLoader,
+)
```

Comment:
For the after migration examples, let's:

1. Add two loader instantiations, one for the file loader and one for the container loader, to mirror the before instantiations
2. Name the variables for those two calls `file_loader` and `container_loader` to make it clear which example matches to which.